packages/ai-core/src/browser/ai-activation-service.ts

```diff
@@ -44,7 +44,8 @@ export class AIActivationService implements FrontendApplicationContribution {
         return this.isAiEnabledKey.get() ?? false;
     }
 
-    initialize(): MaybePromise<void> {
+    async initialize(): Promise<void> {
+        await this.preferenceService.ready;
```

Comment:
@sdirix Is this alright from a performance perspective, or should we rather use `preferenceService.ready.then(() => ...)`?